gnmpagerduty/tasks.py

```diff
@@ -0,0 +1,181 @@
+                                    user=settings.VIDISPINE_USERNAME,passwd=settings.VIDISPINE_PASSWORD,
+                                    run_as=settings.VIDISPINE_USERNAME)
+    sd = {'data':'value', 'test':'value'}
+    sd['map'] = map(lambda (k,v): v.__dict__, storage_data.items())
+
+    returnthis = "Under threshold:"
+
+    url = 'https://events.pagerduty.com/generic/2010-04-15/create_event.json'
+
+    for val in sd['map']:
+        record = StorageData.objects.get(storage_id=val['name'])
+        try:
+            key = IncidentKeys.objects.get(storage_id=val['name'])
+        except Exception as e:
+            key = IncidentKeys.objects.create(storage_id=val['name'])
+
+        intts = int(record.trigger_size)
+        intfc = int(val['freeCapacity'])
+        hffc = human_friendly(intfc)
+        intc = int(val['capacity'])
+        percent_free_float = (100.0 / intc) * intfc
+        percent_free = int(percent_free_float)
+
+        if intfc < intts:
+
+            returnthis = returnthis + " " + val['name']
+            if key.incident_key == '':
+
+                payload = {
+                    "service_key": settings.PAGERDUTY_KEY,
+                    "event_type": "trigger",
+                    "description": "Storage {0} lacks sufficient free space. It is {1}% full.".format(val['contentDict']['name'],percent_free),
+                    "details": {
+                        "Vidispine ID": val['name'],
+                        "Name": val['contentDict']['name'],
+                        "Free Capacity": hffc,
+                        "System Type": system_type
+                        },
+                }
+
+                r = requests.post(url, data=json.dumps(payload))
+
+                pddata = json.loads(r.text)
+
+                key.incident_key = pddata['incident_key']
+                key.save()
+
+        else:
+
+            if key.incident_key != '':
+
+                payload = {
+                    "service_key": settings.PAGERDUTY_KEY,
+                    "event_type": "resolve",
+                    "description": "Storage {0} no longer lacks sufficient free space. It is {1}% full.".format(val['contentDict']['name'],percent_free),
+                    "incident_key": key.incident_key,
+                    "details": {
+                        "Vidispine ID": val['name'],
+                        "Name": val['contentDict']['name'],
+                        "Free Capacity": hffc,
+                        "System Type": system_type,
+                        "Space Used when Resolved": "{0}%".format(percent_free),
+                        },
+                }
+
+                r = requests.post(url, data=json.dumps(payload))
+
+                if r.status_code == 200:
+                    key.incident_key = ''
+                    key.save()
+
+
+    if returnthis is "Under threshold:":
+        returnthis = "No storages are under their thresholds."
```

Comment:
It the value of returnthis has not changed then no storages are under thresholds so change the value of returnthis to a message to that effect. This can then be seen in the Celery output.